### PR TITLE
Models

### PR DESCRIPTION
**Pull request by govza**
Models for Reciter and Recitations.

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -0,0 +1,12 @@
+sudo: required
```
Comment (al-muammar) [inline .travis.yml:1]: It isn't connected with this PR in any way. Please, move it to a separate PR.

File: backend/Dockerfile [hunk 2]
```diff
@@ -16,7 +17,7 @@ RUN apk --no-cache add \
     build-base \
     gettext
 RUN pip3 install --upgrade pip
-RUN pip3 install -r requirements.txt
+RUN pip3 install -r testing.txt
```
Comment (al-muammar) [inline backend/Dockerfile:20]: Why did you decide to rename it?

File: backend/quran/models.py [hunk 3]
```diff
@@ -0,0 +1,40 @@
+'''Models describing abstractions used at Quran app'''
+from django.db import migrations, models
+
+
+class Surah(models.Model):
+    '''General information about Surahs'''
```
Comment (al-muammar) [inline backend/quran/models.py:6]:
```
    """Model representing a surah from the Qur’an."""
```

File: backend/quran/migrations/0001_initial.py [hunk 4]
```diff
@@ -0,0 +1,63 @@
+# Generated by Django 2.1 on 2018-10-14 20:00
```
Comment (al-muammar) [inline backend/quran/migrations/0001_initial.py:1]: Apply formatting to this file.

File: backend/quran/models.py [hunk 5]
```diff
@@ -0,0 +1,40 @@
...
+    surah = models.ForeignKey(
+        Surah,
+        on_delete=models.CASCADE,
+        related_name='ayahs')
+
+
+class Reciter(models.Model):
+    '''General information about Reciter'''
+    name = models.CharField(max_length=100)
+    quality = models.CharField(max_length=50, blank=True)
```
Comment (al-muammar) [inline backend/quran/models.py:25]: Make it a choice filed, decrease the number of characters.
What does the quality signify?

File: backend/pytest.ini [hunk 6]
```diff
@@ -0,0 +1,4 @@
+[pytest]
```
Comment (al-muammar) [inline backend/pytest.ini:1]: Move pytest into another PR.

File: backend/quran/models.py [hunk 7]
```diff
@@ -0,0 +1,40 @@
...
+    '''General information about Ayah'''
+    number = models.PositiveIntegerField()
+    text = models.TextField()
+    surah = models.ForeignKey(
+        Surah,
+        on_delete=models.CASCADE,
+        related_name='ayahs')
+
+
+class Reciter(models.Model):
```
Comment (al-muammar) [inline backend/quran/models.py:22]: I can't see `speed` option. A reciter could recite slowly or quickly. We might add it here.

File: backend/quran/models.py [hunk 8]
```diff
@@ -0,0 +1,40 @@
...
+    '''General information about Reciter'''
+    name = models.CharField(max_length=100)
+    quality = models.CharField(max_length=50, blank=True)
+    description = models.CharField(max_length=255, blank=True)
```
Comment (al-muammar) [inline backend/quran/models.py:26]: What is a use-case for `description`? Can't this information be decoded in other fields?

File: backend/quran/tests.py [hunk 9]
```diff
@@ -0,0 +1,4 @@
+class TestSimple:
```
Comment (al-muammar) [inline backend/quran/tests.py:1]: Move it into a separate diff.

File: backend/testing.txt [hunk 10]
```diff
@@ -0,0 +1,3 @@
+-r requirements.txt
```
Comment (al-muammar) [inline backend/testing.txt:1]: Move it to another PR.

File: docker-compose-dev.yml [hunk 11]
```diff
@@ -0,0 +1,34 @@
+version: "3"
```
Comment (al-muammar) [inline docker-compose-dev.yml:1]: Move it to a separate PR.

File: docker-compose.yml [hunk 12]
```diff
@@ -1,36 +1,14 @@
 version: "3"
 services:
-  web:
-    build: .
-    container_name: web01
-    env_file:
-      - ./.env
-      - ./.env.secret
+  backend:
```
Comment (al-muammar) [inline docker-compose.yml:3]: To another PR.

File: backend/quran/migrations/0002_load_data.py [hunk 13]
```diff
@@ -34,6 +42,30 @@ def unload_ayahs(apps, schema_editor):
...
+    Reciter = apps.get_model("quran", "Reciter")
+    Reciter.objects.all().delete()
+
+
+def load_recitations(apps, schema_editor):
+    for fixture_file in fixture_recitations_list:
+        call_command('loaddata', fixture_file)
+
+
+def unload_recitations(apps, schema_editor):
```
Comment (al-muammar) [inline backend/quran/migrations/0002_load_data.py:62]: You have 4 `unload_*` functions, but they are doing the same work. Merge it into one function.

File: backend/config/testing.py [hunk 14]
```diff
@@ -0,0 +1,8 @@
+from .settings import *
```
Comment (al-muammar) [inline backend/config/testing.py:1]: Move it to a separate PR.

File: backend/helpers/recitation/convert_recitation_csv.py [hunk 15]
```diff
@@ -0,0 +1,106 @@
+'''Convert recit-segments.csv to django fixtures file'''
```
Comment (al-muammar) [inline backend/helpers/recitation/convert_recitation_csv.py:1]: There is no word `recit` in English.

File: backend/quran/migrations/0002_load_data.py [hunk 16]
```diff
@@ -1,13 +1,21 @@
...
 fixture_dir = path.abspath(
     path.join(path.dirname(__file__), '../fixtures/'))
 fixture_surahs = 'surah_list.json'
 fixture_ayahs = 'ayah_list.json'
+fixture_reciters = 'reciters_list.json'
+
+recitation_dir = path.join(fixture_dir, 'recitation')
+
+#  get all recitations json files from fixtures/recitation directory
+fixture_recitations_list = [
```
Comment (al-muammar) [inline backend/quran/migrations/0002_load_data.py:16]: Why don't you put it inside the function?

**Review by al-muammar — CHANGES_REQUESTED** [review]
1. Move the refactoring part and moving to `backend` into a separate PR
2. Move the testing part into a separate PR
3. Write a better description, mention how your data is structured, otherwise, it is difficult to review the PR
4. Give a more descriptive title to your PR. What models? What is happening with them?

**Comment by govza** [comment]
Think this branch is way outdated and I have no desire to update this.
Homework: think about it and understand why.

**Pull request closed without merging**